doc/interfaces/serial.rst

```diff
@@ -13,5 +22,80 @@ Bus
 
 Internals
 ---------
+The frame that will be send and receive over the serial interface consists of
+six parts. The start and the stop byte for the frame, the timestamp, DLC,
+arbitration ID and the payload. Only the payload has a variable length of bytes
+(0 - 8 bytes), the other parts are fix. Both, the timestamp and the arbitration
+ID will be interpreted as 4 byte unsigned integers. The DLC is also a unsigned
+integer with a length of 1 byte.
+
+Serial frame format
+^^^^^^^^^^^^^^^^^^^
++-------------------+----------------+-----------------------------------------------+-------------------------------+-------------------------+---------+--------------+
+|                   | Start of frame | Timestamp                                     | DLC                           | Arbitration ID          | Payload | End of frame |
++===================+================+==============================+================+===============================+=========================+=========+==============+
+| **Length (Byte)** | 1              | 4                                             | 1                             | 4                       | 0 - 8   | 1            |
++-------------------+----------------+-----------------------------------------------+-------------------------------+-------------------------+---------+--------------+
+| **Data type**     | Byte           | Unsigned 4 byte integer                       | Unsigned 1 byte integer       | Unsigned 4 byte integer | Byte    | Byte         |
++-------------------+----------------+-----------------------------------------------+-------------------------------+-------------------------+---------+--------------+
+| **Byte order**    | \-             | Little-Endian                                 | Little-Endian                 | Little-Endian           | \-      | \-           |
++-------------------+----------------+-----------------------------------------------+-------------------------------+-------------------------+---------+--------------+
+| **Description**   | Must be 0xAA   | Usually s, ms or µs since start of the device | Length in byte of the payload | \-                      | \-      | Must be 0xBB |
```

Comment:
As the timestamp could have different scalars for different serial devices (which I'm not entirely sure is reasonable), then we should allow the user to inform the `Serial` class of this scalar to ensure that when serializing/deserializing a frame we apply a correction to the timestamp.

Even if we set some standard (e.g. integer microseconds), there will be some smarter devices (e.g. a raspberry pi) which will actually have epoch time, and some use cases that will want to collect more than `MAX_INT` microseconds. 

@christiansandberg care to weigh in here?